site/content/docs/howto/configure-auth-for-webapps.md

```diff
@@ -0,0 +1,347 @@
+  or not specify the `response_mode` param, which defaults to `query`.
+
+Most web application frameworks offer all these capabilities in their OAuth2/OIDC libraries.
+
+## Create an OIDCClient
+
+For each web application, the administrator of the Pinniped Supervisor will create an OIDCClient describing what
+that web application is allowed to do:
+
+```yaml
+apiVersion: config.supervisor.pinniped.dev/v1alpha1
+kind: OIDCClient
+metadata:
+  # name must have client.oauth.pinniped.dev- prefix
+  name: client.oauth.pinniped.dev-my-webapp-client 
+  namespace: supervisor # must be in the same namespace as the Supervisor
+spec:
+  allowedRedirectURIs:
+    - https://my-webapp.example.com/callback
+  allowedGrantTypes:
+    - authorization_code
+    - refresh_token
+    - urn:ietf:params:oauth:grant-type:token-exchange
+  allowedScopes:
+    - openid
+    - offline_access
+    - pinniped:request-audience
+    - username
+    - groups
+```
+
+If you've saved this into a file `my-oidc-client.yaml`, then install it into your cluster using:
+
+```sh
+kubectl apply -f my-oidc-client.yaml
+```
+
+Do not share OIDCClients between multiple web applications. Each web application should have its own OIDCClient.
+
+The `name` of the OIDCClient will be the client ID used by the web application in the OIDC flows.
+
+The `allowedGrantTypes` and `allowedScopes` decides what the web application is allowed to do with respect to
+authentication. There are several typical combinations of these settings:
+
+1. A web application which is allowed to use the Supervisor for authentication, and furthermore is allowed to
+   authenticate into Kubernetes clusters and perform actions on behalf of the users (using the user's identity):
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+        - urn:ietf:params:oauth:grant-type:token-exchange
+      allowedScopes:
+        - openid
+        - offline_access
+        - pinniped:request-audience
+        - username
+        - groups
+    ```
+
+2. A web application which is allowed to use the Supervisor for authentication, but cannot perform actions on
+   Kubernetes clusters.
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+      allowedScopes:
+        - openid
+        - offline_access
+        - username
+        # "groups" can be excluded from this list when the webapp does
+        # not need to see the group memberships of the users.
+        - groups 
+    ```
+
+3. A web application which is allowed to use the Supervisor for authentication, but cannot see the username or
+   group memberships of the authenticated users, and cannot perform actions on Kubernetes clusters.
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+      allowedScopes:
+        - openid
+        - offline_access
+    ```
+
+## Create a client secret for the OIDCClient
```

Comment:
Also add a new section for deleting an OIDCClient. Note that its client secrets will automatically be deleted when the client is deleted. Deleting an OICClient and then creating it again, even if the exact same name and configuration are used, will result in the new client having no client secrets.